lib/ecom/hooks.ts

```diff
@@ -169,3 +169,39 @@ export const useCart = () => {
         checkout,
     };
 };
+
+export function useCategoryDetails(slug: string): SWRResponse<CollectionDetails> {
+    const ecomApi = useEcomAPI();
+    return useSwr(
+        ['category-details', slug],
+        async ([, slug]) => {
+            const response = await ecomApi.getCategoryBySlug(slug);
+            if (response.status === 'failure') throw response.error;
+            return response.body;
+        },
+        {
+            keepPreviousData: false,
```

Comment:
`keepPreviousData: false` - allows us to switch to skeletons if the user changes the `categorySlug` via Codux to something invalid. Otherwise the component would display the last valid response from the old slug.